File: Classes/TopScoresActivityLevelSelector.m
//
//  TopScoresActivityLevelSelector.m
//  ScoresDisplay
//
//  Created by Jagmeet Chawla on 4/12/09.
//  Revised by Krishna Narayan on 9/17/09
//		--Updated graphics
//		--Added tell a friend
//		--Changed app name to Kids Learn To Tell Time
//  Copyright 2009-2013 NSC Partners LLC. Copyright 2014 One Step Ahead Apps, LLC. All rights reserved.
// Revised by Krishna Narayan on 5/30/26 — Used Claude to migrate to Swift, fix UI Views, remove deprecations, update for iPad, modernize for Apple UI rules.
// Copyright 2026 Island Innovation LLC.  All rights reserved.
//

#import "TopScoresActivityLevelSelector.h"
#import "KidsTimeFunAppState.h"
#import "TopScoresDetailViewController.h"

@implementation TopScoresActivityLevelSelector

@synthesize activity;
@synthesize activityLevel;

- (void)viewDidLoad {
    NSString *activityName;
    switch (self.activity) {
        case kActTellTime:
            activityName = kStrTellTime;
            break;
        case kActElapsedTime:
            activityName = kStrElapsedTime;
            break;
        case kActTimeAfter:
            activityName = kStrTimeAfter;
            break;
        case kActTimeBefore:
            activityName = kStrTimeBefore;
            break;
        case kActSetTime:
            activityName = kStrSetTime;
            break;
        case kActMixed:
            activityName = kStrMixed;
            break;
        default:
            activityName = @"";
            break;
    }
	self.title = [NSString stringWithFormat:@"%@ - %@",kStrTopScores,activityName];
	UIBarButtonItem *homeBarButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"Top Scores"] style:UIBarButtonItemStyleBordered target:self action:nil];
	self.navigationItem.backBarButtonItem = homeBarButton;
	[homeBarButton release];	
	[super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
	// Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
}

- (void)viewDidUnload {
}


#pragma mark Table view methods

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}


// Customize the number of rows in the table view.
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return kNumberOfActivityLevels;
}


// Customize the appearance of table view cells.
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *CellIdentifier = @"Activity";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    if (cell == nil) {
        cell = [[[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:CellIdentifier] autorelease];
		}
    
    // Paint Cell Accessory and Cell Background based on activity
    cell.accessoryView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"RightArrowAccessory"]];
    [cell.textLabel setTextColor:[UIColor whiteColor]];
    switch (self.activity) {
		case kActTellTime:
            [cell setBackgroundColor:[UIColor colorWithRed:.956 green:.423 blue:.109 alpha:0.50]];
			break;
		case kActSetTime:
            [cell setBackgroundColor:[UIColor colorWithRed:.408 green:0.0 blue:.972 alpha:0.50]];
			break;
		case kActTimeAfter:
            [cell setBackgroundColor:[UIColor colorWithRed:.984 green:0.0 blue:.972 alpha:0.50]];
			break;
		case kActTimeBefore:
            [cell setBackgroundColor:[UIColor colorWithRed:.043 green:.808 blue:.11 alpha:0.50]];
			break;
		case kActElapsedTime:
            [cell setBackgroundColor:[UIColor colorWithRed:.043 green:.349 blue:.976 alpha:0.50]];
			break;
		case kActMixed:
            [cell setBackgroundColor:[UIColor colorWithPatternImage:[UIImage imageNamed:@"MixedPattern"]]];
			break;
		default:
			break;
	}
    
	// Configure the cell.
	switch (indexPath.row) {
		case kActLevelYellowBelt:
			cell.imageView.image = [UIImage imageNamed:@"Yellow Belt"];
			cell.textLabel.text = @"Yellow Belt";
			break;
		case kActLevelGreenBelt:
			cell.imageView.image = [UIImage imageNamed:@"Green Belt"];
			cell.textLabel.text = @"Green Belt";
			break;
		case kActLevelRedBelt:
			cell.imageView.image = [UIImage imageNamed:@"Red Belt"];
			cell.textLabel.text = @"Red Belt";
			break;			
		case kActLevelBlackBelt:
			cell.imageView.image = [UIImage imageNamed:@"Black Belt"];
			cell.textLabel.text = @"Black Belt";
			break;
		default:
			break;
	}
	//cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [[tableView cellForRowAtIndexPath:indexPath] setSelectionStyle:UITableViewCellSelectionStyleNone];
	self.activityLevel = (int) indexPath.row;
	TopScoresDetailViewController *topScoreDetailViewController = nil;
	if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) 
	{
		topScoreDetailViewController = [[TopScoresDetailViewController alloc] initWithActivity:self.activity andType:[KidsTimeFunAppState sharedState].activityType andLevel:self.activityLevel showActivityTypeSelection:YES withNibName:@"TopScoresDetailView-iPad" andBundle:nil];
	}
	else {
		topScoreDetailViewController = [[TopScoresDetailViewController alloc] initWithActivity:self.activity andType:[KidsTimeFunAppState sharedState].activityType andLevel:self.activityLevel showActivityTypeSelection:YES withNibName:@"TopScoresDetailView" andBundle:nil];		
	}


	[self.navigationController pushViewController:topScoreDetailViewController animated:YES];
	[topScoreDetailViewController release];
}

- (void)dealloc {
    [super dealloc];
}


@end

File: Classes/TopScoresActivityLevelSelector.swift
import UIKit

@objc(TopScoresActivityLevelSelector)
class TopScoresActivityLevelSelector: UIViewController, UITableViewDelegate, UITableViewDataSource {

    var activity: Int32 = 0
    var activityLevel: Int32 = 0

    override func viewDidLoad() {
        super.viewDidLoad()
        edgesForExtendedLayout = []
        let activityName: String
        switch activity {
        case kActTellTime: activityName = kStrTellTime
        case kActElapsedTime: activityName = kStrElapsedTime
        case kActTimeAfter: activityName = kStrTimeAfter
        case kActTimeBefore: activityName = kStrTimeBefore
        case kActSetTime: activityName = kStrSetTime
        case kActMixed: activityName = kStrMixed
        default: activityName = ""
        }
        title = "\(kStrTopScores) - \(activityName)"
        let btn = UIBarButtonItem(image: UIImage(named: "Top Scores"), style: .plain, target: self, action: nil)
        navigationItem.backBarButtonItem = btn
    }

    func numberOfSections(in tableView: UITableView) -> Int { 1 }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return Int(kNumberOfActivityLevels)
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let id = "Activity"
        let cell = tableView.dequeueReusableCell(withIdentifier: id) ??
            UITableViewCell(style: .default, reuseIdentifier: id)

        cell.accessoryView = UIImageView(image: UIImage(named: "RightArrowAccessory"))
        cell.textLabel?.textColor = .white

        let colors: [Int32: UIColor] = [
            kActTellTime:    UIColor(red: 0.956, green: 0.423, blue: 0.109, alpha: 0.50),
            kActSetTime:     UIColor(red: 0.408, green: 0.0,   blue: 0.972, alpha: 0.50),
            kActTimeAfter:   UIColor(red: 0.984, green: 0.0,   blue: 0.972, alpha: 0.50),
            kActTimeBefore:  UIColor(red: 0.043, green: 0.808, blue: 0.11,  alpha: 0.50),
            kActElapsedTime: UIColor(red: 0.043, green: 0.349, blue: 0.976, alpha: 0.50),
            kActMixed:       UIColor(patternImage: UIImage(named: "MixedPattern")!)
        ]
        cell.backgroundColor = colors[activity] ?? .clear

        switch indexPath.row {
        case Int(kActLevelYellowBelt): cell.imageView?.image = UIImage(named: "Yellow Belt"); cell.textLabel?.text = "Yellow Belt"
        case Int(kActLevelGreenBelt):  cell.imageView?.image = UIImage(named: "Green Belt");  cell.textLabel?.text = "Green Belt"
        case Int(kActLevelRedBelt):    cell.imageView?.image = UIImage(named: "Red Belt");    cell.textLabel?.text = "Red Belt"
        case Int(kActLevelBlackBelt):  cell.imageView?.image = UIImage(named: "Black Belt");  cell.textLabel?.text = "Black Belt"
        default: break
        }
        cell.selectionStyle = .none
        return cell
    }

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        activityLevel = Int32(indexPath.row)
        let nib = UIDevice.current.userInterfaceIdiom == .pad
            ? "TopScoresDetailView-iPad" : "TopScoresDetailView"
        let vc = TopScoresDetailViewController(activity: activity, andType: KidsTimeFunAppState.sharedState().activityType,
                                               andLevel: activityLevel, showActivityTypeSelection: true,
                                               withNibName: nib, andBundle: nil)
        navigationController?.pushViewController(vc, animated: true)
    }
}
